docs/services/deployment/cdk.md

```diff
@@ -0,0 +1,132 @@
+  handler: "com.example.service.package.LambdaHandler", // use the fully qualified name of your handler class
+  timeout: cdk.Duration.seconds(10),
+});
+```
+
+For TypeScript/JavaScript services, you can use the [`NodejsFunction`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_nodejs-readme.html)
+construct to transpile and bundle all the required dependencies.
+
+```typescript
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";
+
+const service = new NodejsFunction(scope, "RestateService", {
+  runtime: lambda.Runtime.NODEJS_18_X,
+  architecture: lambda.Architecture.ARM_64,
+  entry: "handler", // use the path to your service handler
+});
+```
+
+These definitions are no different to how you might deploy any other Lambda handler.
+
+### Register Lambda-based services with Restate
+
+To enable our Lambda-deployed handlers to receive Restate requests, we need to register them with a Restate instance.
+We can model existing Restate Cloud instance using the `RestateCloudEndpoint` construct:
+
+```typescript
+import * as restate from "@restatedev/restate-cdk";
+
+const restateInstance = new restate.RestateCloudEndpoint(scope, "RestateCloud", {
+  clusterId: "restate-cluster-id",
+  authTokenSecretArn: restateAuthToken.secretFullArn,
+});
+```
+
+Automatically register service handlers with Restate on deployment:
+
+```typescript
+import * as restate from "@restatedev/restate-cdk";
+
+const handlers = new restate.LambdaServiceRegistry(scope, "RestateServiceRegistry", {
+  serviceHandlers: {
+    "namespace.Service": service,
+  },
+  restate: restateInstance,
+});
+handlers.register({
+  metaEndpoint: restateInstance.metaEndpoint,
+  invokerRoleArn: restateInstance.invokerRole.roleArn,
+  authTokenSecretArn: restateInstance.authToken.secretArn,
+});
+```
+
+### Deploy a single-node development Restate instance
+
+Restate is [very easy to deploy](https://docs.restate.dev/restate/deployment), and we provide a CDK construct to take
+advantage of this to deploy a standalone self-hosted instance running on Amazon EC2. The `SingleNodeRestateInstance`
+construct deploys a single-node server and creates a new role that can be assumed by the Restate managed service with
+permission to invoke service handlers in your account. The instance will be deployed in a new internet-connected VPC,
+unless you provide one explicitly. You can also optionally enable tracing integration, which will grant the instance
+role permission to send traces to AWS X-Ray.
+
+```typescript
+import * as restate from "@restatedev/restate-cdk";
+
+const restateInstance = new restate.SingleNodeRestateInstance(scope, "RestateServer", {
+  restateTag: "latest",
+  tracing: restate.TracingMode.AWS_XRAY,
+  logGroup: new logs.LogGroup(scope, "RestateLogs", {
+    logGroupName: "/restate/server-logs",
+    retention: logs.RetentionDays.ONE_MONTH,
```

Comment:
Should we also add how to define a specific VPC for deployment here?

Reply:
I left it out because I think it's a better default to deploy Restate in its own VPC. Advanced customers who know what they're doing can easily define their own or connect it to an existing VPC.

A related question is whether we should similarly auto-create a log group for Restate server logs if the user doesn't supply one. Right now that's a required property.